components/sidebar/sidebar_channel/sidebar_channel.tsx

```diff
@@ -215,6 +248,10 @@ export default class SidebarChannel extends React.PureComponent<Props, State> {
             );
         }
 
+        if (!this.state.show) {
```

Comment:
Could you move this up closer to the top of the render? I know it won't make much of a difference to skip the whole setup of `wrappedComponent`, but it's nice to return early when rendering null doesn't depend on any of the other work done by this function.

Reply:
There were few UI gliches when we werent rendering the draggable components so i moved this check above https://github.com/mattermost/mattermost-webapp/pull/9732/files#diff-4c9f89f66f73c204771dc4e66a58fb0c67b346703ea3753e8c3337a99e9f9540R175

Reply:
Okay, that should be fine. I assume the heaviest part of those extra channels in the sidebar is the actual `channelComponent`, so that should still be a performance improvement. I will have to update my queries I've been asking people to run since they'll still count those empty draggable items